docs/source/plugin/what.rst

```diff
@@ -0,0 +1,133 @@
+=================
+What is a Plugin?
+=================
+
+Before writing your first plugin, you may want to know what is a plugin, how is
+it made, and what are the best solutions for your need. In this chapter we'll
+cover what is a Plugin, what you can do with them, and some vocabulary.
+
+You may want to skip ahead to :doc:`anatomy` if you already know what a Plugin
+is and what types of plugin there are.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Plugin Types
+============
+
+There are four types of plugins: :term:`Single file plugin`,
+:term:`Folder plugin`, :term:`Namespace package plugin`, and
+:term:`Entry point plugin`.
+
+Single file
+-----------
+
+A **Single file** plugin is the most basic form a Sopel plugin can take. It is
+composed of a single file available to Sopel from its plugin's directory (or
```

Comment:
```suggestion
composed of a single file available to Sopel from its ``plugins`` directory (or
```

?

Reply:
I wanted to say: the directory where Sopel looks for plugins.

Reply:
Oh, maybe just `plugins` without the backticks/formatting then.

Reply:
```suggestion
composed of a single file available to Sopel from its plugins directory (or
```